File: web_auto/automation_java/src/main/java/pages/LoginPage.java
package pages;

import base.BasePage;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

/**
 * Page Object for Login page.
 * Contains locators and methods specific to login functionality.
 */
public class LoginPage extends BasePage {

    // Page locators using @FindBy annotations
    @FindBy(id = "username")
    private WebElement usernameField;

    @FindBy(id = "password")
    private WebElement passwordField;

    @FindBy(id = "login-button")
    private WebElement loginButton;

    @FindBy(className = "error-message")
    private WebElement errorMessage;

    @FindBy(className = "success-message")
    private WebElement successMessage;

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    /**
     * Enter username
     */
    public void enterUsername(String username) {
        sendKeysToElement(usernameField, username);
    }

    /**
     * Enter password
     */
    public void enterPassword(String password) {
        sendKeysToElement(passwordField, password);
    }

    /**
     * Click login button
     */
    public void clickLoginButton() {
        clickElement(loginButton);
    }

    /**
     * Perform complete login action
     */
    public void login(String username, String password) {
        enterUsername(username);
        enterPassword(password);
        clickLoginButton();
    }

    /**
     * Get error message text
     */
    public String getErrorMessage() {
        return getElementText(errorMessage);
    }

    /**
     * Check if error message is displayed
     */
    public boolean isErrorMessageDisplayed() {
        return isElementDisplayed(errorMessage);
    }

    /**
     * Check if login was successful
     */
    public boolean isLoginSuccessful() {
        return isElementDisplayed(successMessage);
    }
}

File: web_auto/automation_js/pages/loginPage.js
const { expect } = require('@playwright/test');
const BasePage = require('./BasePage');

/**
 * Page Object Model for Login page
 * Encapsulates login page elements and actions
 */
class LoginPage extends BasePage {
  constructor(page) {
    super(page);

    // Page elements
    this.usernameField = page.locator('#username');
    this.passwordField = page.locator('#password');
    this.loginButton = page.locator('#login-button');
    this.errorMessage = page.locator('.error-message');
    this.successMessage = page.locator('.success-message');
  }

  /**
   * Navigate to login page
   */
  async navigate() {
    await this.page.goto('/index.html');
  }

  /**
   * Enter username
   */
  async enterUsername(username) {
    await this.safeFill(this.usernameField, username);
  }

  /**
   * Enter password
   */
  async enterPassword(password) {
    await this.safeFill(this.passwordField, password);
  }

  /**
   * Click login button
   */
  async clickLoginButton() {
    await this.safeClick(this.loginButton);
  }

  /**
   * Perform complete login action
   */
  async login(username, password) {
    await this.enterUsername(username);
    await this.enterPassword(password);
    await this.clickLoginButton();
  }

  /**
   * Get error message text
   */
  async getErrorMessage() {
    return await this.errorMessage.textContent();
  }

  /**
   * Check if error message is visible
   */
  async isErrorMessageVisible() {
    return await this.errorMessage.isVisible();
  }

  /**
   * Check if login was successful
   */
  async isLoginSuccessful() {
    return await this.successMessage.isVisible();
  }

  /**
   * Wait for login page to load
   */
  async waitForPageLoad() {
    await this.page.waitForLoadState('networkidle');
    await expect(this.usernameField).toBeVisible();
  }
}

module.exports = LoginPage;
